### ISSUE TITLE
Assigning a file key while some volumes are full

### ISSUE THREAD

**Issue by SakaRawks** [chrislusf/seaweedfs#244]
Hello guys,

We're currently testing Seaweed for a possible future use in a file hosting solution.

We have 3 master servers, and 3 volumes servers ATM.

On each of these 3 volume servers, we have 6 volumes.

3 of these 6 volumes are full, and the remaining 3 are empty.

The problem is, when we send a GET request to /dir/assign, it keeps on returning us file keys that are linked to the full volumes. The "Volume Layout" structure's "Writable" attribute still contains the full volumes when they shouldn't be there. Logically, it should only contain volumes 4, 5 and 6 but it currently has 1, 2, 3, 4, 5 and 6, making that half of our file keys assignations are leading to an error when uploading because the volume is full.

What is the condition for a volume not to be included in the writable volumes array?

My guess is that the volume still has some bytes free, so it stays writable, but the free space isn't large enough to upload any normal file. Could I be right ?

Any solution for this ?

Thanks in advance.

Post scriptum: Another question: what amount of data is taken in memory for a single 10 MB file ? I've read in your "Optimization" page that 1.5 Million 20 KB files are taking 24bytes per entry in memory, making it 36 MB per volume. What would be the values for 3k 10MB files ? (KB/file)

Is there a formula to calculate the memory needed per volume server in terms of average file size?

**Comment by chrislusf** [comment]
Please share the screenshot of http://master:9333/ to see the details of the volumes.

**Comment by SakaRawks** [comment]
We just fixed it, had a wrong -volumeSizeLimitMB when starting the volume server. It was over the theoretical maximum size of a volume which is 35 GB, and when we assign a file key it considers the volume as not full because it is not reaching the value we passed.

Sorry for the long post.

For now SeaweedFS seems to be a pretty robust solution, congratulations for your hard work on it.

Any answer to my second question? 

Thanks again.
